Framework/Geometry/inc/MantidGeometry/Instrument.h

```diff
@@ -327,6 +327,10 @@ class MANTID_GEOMETRY_DLL Instrument : public CompAssembly {
 
   /// Pointer to the reference frame object.
   boost::shared_ptr<ReferenceFrame> m_referenceFrame;
+
+  /// Pointer to the DetectorInfo object. NULL unless the instrument is
+  /// associated with an ExperimentInfo object.
+  const Beamline::DetectorInfo *m_detectorInfo{nullptr};
```

Comment:
Eventually everything: `API::DetectorInfo` will be removed. As first steps it will be masking, and then position information.